src/test/regress/expected/foreign_tables_mx.out

```diff
@@ -182,6 +182,19 @@ ALTER USER MAPPING FOR postgres SERVER foreign_server OPTIONS (SET user 'nonexis
 SELECT * FROM foreign_table ORDER BY a;
 ERROR:  could not connect to server "foreign_server"
 ALTER USER MAPPING FOR postgres SERVER foreign_server OPTIONS (SET user 'postgres');
+-- test truncate
+TRUNCATE foreign_table;
+\c - - - :worker_1_port
+SET search_path TO foreign_tables_schema_mx;
+-- verify the table is truncated
```

Comment:
It seems like `TRUNCATE` from the worker fails as the user mapping does not exists.

```SQL
truncate foreign_table;
ERROR:  user mapping not found for "onderkalaci"
```

Where the bt is the following where the standardProcessUtility fails with lack of user mapping.
```
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 7.1
  * frame #0: 0x00000001003d41d6 postgres`errstart(elevel=21, domain=0x0000000000000000) at elog.c:406:6
    frame #1: 0x00000001004a5be9 postgres`errstart_cold(elevel=<unavailable>, domain=<unavailable>) at elog.c:326:9
    frame #2: 0x000000010046b8a2 postgres`GetUserMapping.cold.1 at foreign.c:218:3
    frame #3: 0x00000001001beeb5 postgres`GetUserMapping(userid=<unavailable>, serverid=<unavailable>) at foreign.c:218:3
    frame #4: 0x0000000100c0cbe0 postgres_fdw.so`postgresExecForeignTruncate(rels=<unavailable>, behavior=DROP_RESTRICT, restart_seqs=false) at postgres_fdw.c:3021:9
    frame #5: 0x0000000100150cb7 postgres`ExecuteTruncateGuts(explicit_rels=0x00007feeee81d498, relids=<unavailable>, relids_logged=0x0000000000000000, behavior=DROP_RESTRICT, restart_seqs=false) at tablecmds.c:2017:5
    frame #6: 0x000000010015047d postgres`ExecuteTruncate(stmt=0x00007feeee81d368) at tablecmds.c:1708:2
    frame #7: 0x00000001002e6ea3 postgres`standard_ProcessUtility(pstmt=0x00007feeee81d398, queryString="truncate foreign_table;", readOnlyTree=<unavailable>, context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=0x00007feeed80f5a0, qc=0x00007ff7bfefb0d0) at utility.c:732:4
    frame #8: 0x000000010120d266 citus.so`ProcessUtilityInternal(pstmt=0x00007feeee81d398, queryString="truncate foreign_table;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=<unavailable>, completionTag=0x00007ff7bfefb0d0) at utility_hook.c:630:3
    frame #9: 0x000000010120cbc8 citus.so`multi_ProcessUtility(pstmt=0x00007feeed80f4c0, queryString="truncate foreign_table;", readOnlyTree=<unavailable>, context=<unavailable>, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=0x00007feeed80f5a0, completionTag=0x00007ff7bfefb0d0) at utility_hook.c:287:3
    frame #10: 0x0000000101297051 citus.so`ColumnarProcessUtility(pstmt=0x00007feeed80f4c0, queryString="truncate foreign_table;", readOnlyTree=<unavailable>, context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=0x00007feeed80f5a0, completionTag=0x00007ff7bfefb0d0) at columnar_tableam.c:2110:2
    frame #11: 0x0000000100b5dbc0 pg_stat_statements.so`pgss_ProcessUtility(pstmt=0x00007feeed80f4c0, queryString=<unavailable>, readOnlyTree=<unavailable>, context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, queryEnv=0x0000000000000000, dest=0x00007feeed80f5a0, qc=0x00007ff7bfefb0d0) at pg_stat_statements.c:1131:5
    frame #12: 0x00000001002e6ad1 postgres`ProcessUtility(pstmt=0x00007feeed80f4c0, queryString=<unavailable>, readOnlyTree=<unavailable>, context=<unavailable>, params=<unavailable>, queryEnv=<unavailable>, dest=0x00007feeed80f5a0, qc=0x00007ff7bfefb0d0) at utility.c:523:3
    frame #13: 0x00000001002e67d2 postgres`PortalRunUtility(portal=0x00007feeef82fd18, pstmt=0x00007feeed80f4c0, isTopLevel=<unavailable>, setHoldSnapshot=false, dest=0x00007feeed80f5a0, qc=0x00007ff7bfefb0d0) at pquery.c:1155:2
    frame #14: 0x00000001002e5fa8 postgres`PortalRunMulti(portal=0x00007feeef82fd18, isTopLevel=true, setHoldSnapshot=false, dest=0x00007feeed80f5a0, altdest=0x00007feeed80f5a0, qc=0x00007ff7bfefb0d0) at pquery.c:0
```

I think is is still OK to throw an error `If( TruucateOnCitusLocalTable && !IsCoordinator()) error: TRUNCATE to Citus local table can be done from the coordinator`

and, in the source code, mention that user mapping is not propagated by Citus, hence we have this limitation